src/api/api.js

```diff
@@ -10,4 +10,49 @@ export const getBookById = (id) => {
     return new Promise((res, rej) => {
         res(mockBook);
     })
-}
\ No newline at end of file
+}
+
+export const getBooks = (page, genres, sortBy) => {
+    const booksPerPage = 5;
+    let filteredBooks;
+    if(genres !== null && genres.length !== 0) {
+        filteredBooks = mockBooks.filter((book)=>{
+            let filterGenres = book.genres.filter(value => genres.includes(value));
+            return (filterGenres !== null && filterGenres.length !== 0)
+        });
+    }
+    else {
+        filteredBooks = [...mockBooks];
+    }
```

Comment:
```suggestion
    let filteredBooks = [...mockBooks]
    if (genres && genres.length) {
        filteredBooks = mockBooks.filter((book) => {
            return book.genres.some(g => genres.includes(g))
        });
    }
```
- no need for else clause here
- when you're checking things like `x !== null` or `x.length !== 0` you can almost always shorten it to `x && x.length`. Only if, for instance, you want to allow the value to be `undefined` but not `null` you would need to put `!== null`. Otherwise for these 'falsy' values we shorten conditionals.
- you complicated a bit the filtering logic. When it comes to these array manipulations I tend to google search just in case there is a simpler, cleaner way. For this, i googled 'js array includes any of another array' since that is basically what you are checking here, and found [this](https://stackoverflow.com/questions/16312528/check-if-an-array-contains-any-element-of-another-array-in-javascript)
- fixed some spacing issues - you have to let the code 'breathe', it makes it much more readable :)